### DISCUSSION TITLE
Is there a way not to log body if it's binary data?

### DISCUSSION THREAD

**Discussion by vkuznetsovgn** [litestar-org/litestar#2968] [Q&A] [answered]
Hi everyone.

I am using a simple `LoggingMiddlewareConfig` to log requests and responses. Sometimes I have files in requests or responses and I don't want to log the body that consists of binary array. Can I maybe say somehow not to log files or maybe at least set up a size limit for the body to log?

**Comment by peterschutt** [comment, marked as answer]
This is not tested, but should work - subclass `LoggingMiddleware` and inspect/manipulate the data to be logged from there, e.g.:

```py
from typing import Any

from litestar.middleware import LoggingMiddleware, LoggingMiddlewareConfig
from litestar.types import Scope


class MyLoggingMiddleware(LoggingMiddleware):

    def should_remove_body(body: Any) -> bool:
        # custom logic here

    def extract_response_data(scope: Scope) -> dict[str, Any]:
        data = super().response_data(scope)
        if not self.should_remove_body(data.get("body", None)):
            del data["body"]
        return data


config = LoggingMiddlewareConfig(..., middleware_class=MyLoggingMiddleware)
```
